docs/graphql/manual/guides/postgres/constraints.rst

```diff
@@ -0,0 +1,116 @@
+.. meta::
+   :description: Use Postgres constraints with Hasura
+   :keywords: hasura, docs, postgres, constraints
+
+.. _postgres_constraints:
+
+Postgres constraints
+====================
+
+.. contents:: Table of contents
+  :backlinks: none
+  :depth: 2
+  :local:
+
+Introduction
+------------
+
+`Postgres constraints <https://www.postgresql.org/docs/current/ddl-constraints.html>`__ are used to define rules for fields in a database table. They ensure that
+no invalid data is entered into the database.
+
+.. note::
+
+  For more detailed information on Postgres constraints, please refer to the `Postgres documentation <https://www.postgresql.org/docs/current/ddl-constraints.html>`__.
+
+Postgres constraints
+--------------------
+
+There are different types of constraints that can be used with Postgres.
+
+Primary key constraints
+^^^^^^^^^^^^^^^^^^^^^^^
+
+A ``PRIMARY KEY`` is used to identify each specific row of a table.
+
+**Identify the author's id as the primary key of the authors table:**
+
+.. code-block:: sql
+  :emphasize-lines: 2
+
+  CREATE TABLE authors(
+    id INT PRIMARY KEY     NOT NULL,
+    name           TEXT    NOT NULL
+  );
+
+Foreign key constraints
+^^^^^^^^^^^^^^^^^^^^^^^
+
+A foreign key constraint specifies that the values in a column must match the values appearing in a row of another table. 
+Foreign key constraints are used to create relationships between tables.
+
+**Define the author_id in the articles table as a foreign key to the id field in the authors table:**
+
+.. code-block:: sql
+  :emphasize-lines: 5
+
+  CREATE TABLE articles(
+    id SERIAL PRIMARY KEY  NOT NULL,
+    title          TEXT    NOT NULL,
+    author_id INTEGER,
+    FOREIGN KEY (author_id) REFERENCES authors (id)
```

Comment:
should probably show the `authors` table definition as well?

Reply:
Done